File: 2019/2019day1.py
# -*- coding: utf-8 -*-
"""
@author: Aaron
"""
#%%
print("----------------------------")
print("Advent of Code 2019, Day 1")
print("----------------------------")
#%%
import sys
import numpy as np

if len(sys.argv) == 1:
    file = 'input_day1.txt'
else:
    file = sys.argv[1]

with open(file,'r') as f:
    lines = f.readlines()
    
lines = [line.replace('\n','') for line in lines]

series = np.array(lines, dtype=int)

part1_ans = sum(series // 3 - 2)
print("Answer to Part One: " + str(part1_ans))
#%% PART TWO
total = 0
for num in series:
    fuel = num
    while True:
        fuel = fuel // 3 - 2
        if fuel <= 0:
            break
        total += fuel
print("Answer to Part Two: " + str(total))

File: 2019/2019day1.R
args = commandArgs(trailingOnly=TRUE)
if(length(args) == 0) {
  file <- 'input_day1.txt'
} else {
    file <- args[1]}

# Read in File ------------------------------------------------------------

data <- read.csv(file, header=FALSE)

# Part One ----------------------------------------------------------------

total1 <- sum(floor(data / 3) - 2)
print(paste("Answer to Part One:", total1))

# Part Two ----------------------------------------------------------------

# Function to recursively calculate fuel needed, given a number
recursive_fuel <- function(number) {
  total = 0
  number <- floor(number / 3) - 2
  if(number <= 0){return(0)
  } else {
    total <- total + number
    total <- total + recursive_fuel(number)
  }
  return(total)
}

# Apply the function to each number, then take the sum
total2 <- sum(apply(data, MARGIN=1, FUN=recursive_fuel))

print(paste("Answer to Part Two:",total2))
